helix-term/src/ui/statusline.rs

```diff
@@ -205,23 +207,40 @@ where
     );
 }
 
-fn render_diagnostics<F>(context: &mut RenderContext, write: F)
-where
-    F: Fn(&mut RenderContext, String, Option<Style>) + Copy,
-{
-    let (warnings, errors) = context
-        .doc
-        .diagnostics()
-        .iter()
+/// Gets the count pair (warnings, error) of warnings, error on document.
+fn diagnostics_count(doc: &Document) -> (i32, i32) {
+    doc.diagnostics().iter().fold((0, 0), |mut counts, diag| {
+        use helix_core::diagnostic::Severity;
+        match diag.severity {
+            Some(Severity::Warning) => counts.0 += 1,
+            Some(Severity::Error) | None => counts.1 += 1,
+            _ => {}
+        }
+        counts
+    })
+}
+
+/// Gets the count pair (warnings, error) of warnings, error on workspace.
+fn workspace_diagnostic_count(editor: &Editor) -> (i32, i32) {
```

Comment:
These can be inlined into `render_diagnostics` and `render_workspace_diagnostics` respectively

Reply:
Done